### ISSUE TITLE
Invalid default generator "Visual Studio 15 2017 Win64 Win64"

### ISSUE THREAD

**Issue by karel-burda** [cmake-js/cmake-js#160]
Invalid generator is selected with version **5.1.0** on Windows with VS 2017 (double "Win64" appears in the name) by default.

Plain "cmake-js compile" was executed.

**Error output:**

```
c:\burda\projects\wui\tools\com-wui-framework-resource-writer>npm install

resource-writer@0.9.1 install c:\burda\projects\wui\tools\com-wui-framework-resource-writer
cmake-js compile

[ 'C:\\burda\\projects\\wui\\com-wui-framework-builder\\dependencies\\nodejs\\node.exe',
  'c:\\burda\\projects\\wui\\tools\\com-wui-framework-resource-writer\\node_modules\\cmake-js\\bin\\cmake-js',
  'compile' ]
info TOOL Using Visual Studio 15 2017 Win64 Win64 generator.
info CMD CONFIGURE
info RUN cmake "c:\burda\projects\wui\tools\com-wui-framework-resource-writer" --no-warn-unused-cli -G"Visual Studio 15 2017 Win64 Win64" -DCMAKE_JS_VERSION="5.1.0" -DCMAKE_BUILD_TYPE="Release" -DCMAKE_RUNTIME_OUTPUT_DIRECTORY="c:\burda\projects\wui\tools\com-wui-framework-resource-writer\build" -DCMAKE_JS_INC="C:\Users\nxf45876\.cmake-js\node-x64\v10.15.0\include\node" -DNODE_RUNTIME="node" -DNODE_RUNTIMEVERSION="10.15.0" -DNODE_ARCH="x64" -DCMAKE_JS_LIB="C:\Users\nxf45876\.cmake-js\node-x64\v10.15.0\win-x64\node.lib"
Not searching for unused variables given on the command line.
CMake Error: Could not create named generator Visual Studio 15 2017 Win64 Win64

(list of available generators)

```

**Comment by pcristofanilli** [comment]
hi, really newby in compilation.
I've installed
npm -g install cmake-js
I've applied the patch to the file, after this i've installed grunt and  typed:
npm run-script compile
but i don't see changes.
surerly i've missed something.

**Comment by PaulFreund** [comment]
I can reproduce this error and can verify that the pull request fixes the issue. Also without the patch Win64 is also appended for ia32 builds and the wrong generator is choosen with MSVC.

**Comment by pcristofanilli** [comment]
> hi, really newby in compilation.
> I've installed
> npm -g install cmake-js
> I've applied the patch to the file, after this i've installed grunt and typed:
> npm run-script compile
> but i don't see changes.
> surerly i've missed something.
> I've triend also inserting some logs, but nothing happen.
Sorry discovered the mistake, the library the I was trying to install  download his own copy, so the global copy was bypassed.
Thanx to all.
There is a prevision about the release of this fix?

**Comment by BotellaA** [comment]
Maybe CMake changed its output. It looks like `cmake-js` is getting its generator list from this command:
```
cmake  -E capabilities
{"generators":[{"extraGenerators":["CodeBlocks","CodeLite","Sublime Text 2","Kate","Eclipse CDT4"],"name":"MinGW Makefiles","platformSupport":false,"toolsetSupport":false},{"extraGenerators":[],"name":"Visual Studio 12 2013 ARM","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 14 2015 ARM","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 16 2019","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 15 2017 Win64","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 15 2017","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 15 2017 ARM","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 14 2015 Win64","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 12 2013","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 12 2013 Win64","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 14 2015","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 11 2012","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 11 2012 ARM","platformSupport":true,"toolsetSupport":true},{"extraGenerators":["CodeBlocks","CodeLite","Sublime Text 2","Kate","Eclipse CDT4"],"name":"Unix Makefiles","platformSupport":false,"toolsetSupport":false},{"extraGenerators":[],"name":"Visual Studio 11 2012 Win64","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 10 2010 Win64","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 10 2010","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 10 2010 IA64","platformSupport":true,"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 9 2008","platformSupport":true,"toolsetSupport":false},{"extraGenerators":[],"name":"Visual Studio 9 2008 Win64","platformSupport":true,"toolsetSupport":false},{"extraGenerators":["CodeBlocks"],"name":"NMake Makefiles JOM","platformSupport":false,"toolsetSupport":false},{"extraGenerators":[],"name":"Visual Studio 9 2008 IA64","platformSupport":true,"toolsetSupport":false},{"extraGenerators":[],"name":"MSYS Makefiles","platformSupport":false,"toolsetSupport":false},{"extraGenerators":[],"name":"Borland Makefiles","platformSupport":false,"toolsetSupport":false},{"extraGenerators":["CodeBlocks","CodeLite","Sublime Text 2","Kate","Eclipse CDT4"],"name":"NMake Makefiles","platformSupport":false,"toolsetSupport":false},{"extraGenerators":[],"name":"Green Hills MULTI","platformSupport":true,"toolsetSupport":true},{"extraGenerators":["CodeBlocks","CodeLite","Sublime Text 2","Kate","Eclipse CDT4"],"name":"Ninja","platformSupport":false,"toolsetSupport":false},{"extraGenerators":[],"name":"Watcom WMake","platformSupport":false,"toolsetSupport":false}],"serverMode":true,"version":{"isDirty":false,"major":3,"minor":14,"patch":0,"string":"3.14.0","suffix":""}}
```

Maybe you should check only the generator names without architecture information.

**Comment by unbornchikken** [comment]
Fixed as of cmake-js@5.1.1.
